### PR TITLE
#5: Fixed JavaDoc and code smells

### LINKED ISSUE #5: Fix JavaDoc and code smells
## Situation

The UDF API is quite old. While it is stable, there is missing JavaDoc and there are code smells that need fixing.

## Acceptance Criteria

* JavaDoc clean
* Code smells fixed.

### PR DESCRIPTION
**Pull request by redcatbear**
Closes #5.

### CODE REVIEW

File: src/main/java/com/exasol/ExaIterator.java [hunk 1]
```diff
@@ -5,193 +5,281 @@
...
-     * @return value of the specified column of the current row.
-     *         This can be used for all data types. You have to cast the value appropriately.
+     * Get the value of the column with the given name as {@link Object}.
+     * <p>
+     * This can be used for all SQL data types. You have to cast the value appropriately.
+     * </p>
      *
      * @param name name of the column
+     *
+     * @return value of the specified column of the current row as a {@link Object} object.
```
Comment (kaklakariada) [inline src/main/java/com/exasol/ExaIterator.java:282]:
```suggestion
     * @return value of the specified column of the current row as a {@link Object} object or {@code null} if the column value is {@code NULL}
```

File: src/main/java/com/exasol/ExaMetadata.java [hunk 2]
```diff
@@ -3,206 +3,296 @@
...
+     * 
      * @return name of the script scope user
      */
     public String getScopeUser();
 
-
     /**
+     * If the script is executed in the context of a selected (aka. "open") schema, get the name of that schema.
+     * 
      * @return name of the current open schema
```
Comment (kaklakariada) [inline src/main/java/com/exasol/ExaMetadata.java:64]:
```suggestion
     * @return name of the current open schema or {@code null} if no schema is selected
```

File: src/main/java/com/exasol/ExaMetadata.java [hunk 3]
```diff
@@ -3,206 +3,296 @@
...
 
     /**
-     * @return Memory limit for the current JVM process in bytes. If this memory
-     * is exceeded, the database resource management will kill the JVM process.
+     * Get the memory limit for the JVM process that powers the script.
+     * <p>
+     * If this memory is exceeded, the database resource management will kill the JVM process.
+     * </p>
+     * 
+     * @return Memory limit for the current JVM process in bytes.
```
Comment (kaklakariada) [inline src/main/java/com/exasol/ExaMetadata.java:119]:
```suggestion
     * @return memory limit for the current JVM process in bytes.
```

File: src/main/java/com/exasol/ExaMetadata.java [hunk 4]
```diff
@@ -3,206 +3,296 @@
...
      */
     public BigInteger getMemoryLimit();
 
     /**
+     * Get the type of the UDF script.
+     * 
      * @return input type of the script, either "SCALAR" or "SET"
```
Comment (kaklakariada) [inline src/main/java/com/exasol/ExaMetadata.java:126]:
```suggestion
     * @return input type of the script, either {@code SCALAR} or {@code SET}
```

File: src/main/java/com/exasol/ExaMetadata.java [hunk 5]
```diff
@@ -3,206 +3,296 @@
...
-     *         data from this column using ExaIterator.getString(column).
+     * Get the Java data type of the input column at the given index.
+     * <p>
+     * If this method for example returns {@code java.lang.String}, you can access the data from this column using
+     * {@link ExaIterator#getString(int)}.
+     * </p>
+     * 
+     * @param column index of the column, starting with 0
+     * 
+     * @return Java class used to represent data from the specified column. .
```
Comment (kaklakariada) [inline src/main/java/com/exasol/ExaMetadata.java:155]:
```suggestion
     * @return Java class used to represent data from the specified column
```

File: src/main/java/com/exasol/ExaMetadata.java [hunk 6]
```diff
@@ -3,206 +3,296 @@
...
     /**
-     * @param column id of the column, starting with 0
+     * Get the Java data type of the output column at the given index.
+     * <p>
+     * If this returns {@code java.lang.String}, you can emit data for this column using a {@link String}.
+     * </p>
+     *
+     * @param column index of the column, starting with 0
+     *
      * @return the java class used to represent data from the specified output column.
```
Comment (kaklakariada) [inline src/main/java/com/exasol/ExaMetadata.java:229]:
```suggestion
     * @return the java class used to represent data from the specified output column
```

File: src/main/java/com/exasol/ExaMetadata.java [hunk 7]
```diff
@@ -3,206 +3,296 @@
...
+     * class.
+     * <p>
+     * You can use Java Reflection to work with this class.
+     * </p>
+     * <p>
+     * Please note that there is a simple way to include other code using the keywords {@code %import} and {@code %jar}
+     * in the script code (see user manual).
+     * </p>
+     * 
      * @param name The name of the script to be imported (context-sensitive)
```
Comment (kaklakariada) [inline src/main/java/com/exasol/ExaMetadata.java:283]:
```suggestion
     * @param name the name of the script to be imported (context-sensitive)
```
What does "context-sensitive" mean? Maybe a typo -> "case-sensitive"?

**Review by kaklakariada — CHANGES_REQUESTED** [review]

**Comment by redcatbear** [comment]
Sounds reasonable. Changed.

File: src/main/java/com/exasol/ExaConnectionAccessException.java [hunk 8]
```diff
@@ -1,18 +1,30 @@
...
+
+    /**
+     * Create a new instance of an {@link ExaConnectionAccessException}.
+     *
+     * @param message error message
+     */
+    public ExaConnectionAccessException(final String message) {
+        super("connection exception: " + message);
+        this.message = message;
     }
```
[the remark this answers is not in the document]
Comment (redcatbear) [reply src/main/java/com/exasol/ExaConnectionAccessException.java:19]: Let's not touch that out of backward compatibility at the moment.

**Review by tkilias — CHANGES_REQUESTED** [review]

**Review by kaklakariada — APPROVED** [review]

**Pull request merged**